### DISCUSSION TITLE
Road Network data input for Optimisation problems

### DISCUSSION THREAD

**Discussion by rajesvariparasa** [pysal/spaghetti#554] [Q&A] [answered]
What do you think about the quality of [Geofabrik's](https://download.geofabrik.de/asia.html) road network data as input to the model (or any network analysis model in general)? Are there are any automated ways of assessing (maybe quantifying/describing the amount of error?) and improving the quality of a large street network data? Since the analysis has to be carried out at the city level, it's very difficult to build a road network from scratch. And I am looking for ways to minimize the error due to a faulty road network (e.g. lines which run parallel to both sides of a road disconnecting the two sides of the network, absent nodes and overshoots/undershoots instead).

I recently came across [pyrosm](https://pyrosm.readthedocs.io/en/latest/) which outputs graphs for network analysis for OSM data downloaded from providers like Geofabrik, BBBike. Can this help in arriving at a more connected road network with OSM data? And if so, will spaghetti take a network as input as opposed to a raw .shp for the optimisation problems?

**Comment by jGaboardi** [comment, marked as answer]
@rajesvariparasa To be clear, no mathematical optimization actually happens in `spaghetti`. The [optimal facility location](https://pysal.org/spaghetti/notebooks/facility-location.html) tutorial (along with the [traveling salesperson problem](https://pysal.org/spaghetti/notebooks/tsp.html) and [transportation problem](https://pysal.org/spaghetti/notebooks/transportation-problem.html)) are examples of using products generated from `spaghetti` as input for a subsequent problem (origin-destination distance matrices in all cases).

I have no experience with Geofabrik, so I can't speak to that, but you my be interested in `osmnx` for extracting cleaned networks/graphs from OSM data.
* [GitHub](https://github.com/gboeing/osmnx)
* [docs](https://geoffboeing.com/publications/osmnx-complex-street-networks/)
* **Boeing, G.** 2017. “OSMnx: New Methods for Acquiring, Constructing, Analyzing, and Visualizing Complex Street Networks.” Computers, Environment and Urban Systems. 65, 126-139. doi:[10.1016/j.compenvurbsys.2017.05.004](https://www.sciencedirect.com/science/article/pii/S0198971516303970?via%3Dihub)
